src/prefect/futures.py

```diff
@@ -150,6 +162,16 @@ def result(
             _result = run_coro_as_sync(_result)
         return _result
 
+    def add_done_callback(self, fn: Callable[[PrefectFuture], None]):
```

Comment:
maybe we could have a docstring here to briefly explain the `call_with_self` business here?